### PR TITLE
Implemented history feature

### PR DESCRIPTION
**Pull request by ishitakeshawani**
build history page
build a single video page
Implemented add to history, clear history, remove the video from history.

### CODE REVIEW

**Review by surbhi84 — APPROVED** [review]
The code looks neat.
Good naming conventions followed.
I didn't see any breaking changes looks good to merge.

**Review by anupkgurung — APPROVED** [review]
looks good.
You could have used uuid for generating id in video.js

File: src/components/sidebar/SideBar.js [hunk 1]
```diff
@@ -18,15 +18,15 @@ export function SideBar() {
           <i class="fa-solid fa-play"></i>
```
Comment (Adarsh-Patel-Dev) [inline src/components/sidebar/SideBar.js]: use ReactIcons and change class to className

File: src/components/sidebar/SideBar.js [hunk 2]
```diff
@@ -18,15 +18,15 @@ export function SideBar() {
...
           <span>Playlists</span>
         </NavLink>
-        <NavLink className="link-no-style sidebar-item" to="/">
+        <NavLink className="link-no-style sidebar-item" to="/watchlater">
           <i class="fa-solid fa-clock"></i>
```
Comment (Adarsh-Patel-Dev) [inline src/components/sidebar/SideBar.js]: same here

File: src/pages/history/History.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+              <div
+                className="card-section regular-font-weight"
+                id="card-section"
+              >
+                <Link to={`/video/${video._id}`}>
+                  <img className="card-img" src={video.image} alt="thumbnail" />
+                </Link>
+                <div className="card-header">
+                  <div className="card-header-title bold-font-weight">
+                    {" "}
```
Comment (Adarsh-Patel-Dev) [inline src/pages/history/History.js]: remove this

**Review by Adarsh-Patel-Dev — COMMENTED** [review]
Suggested some changes 
rest code is clean & well structured

**Comment by ishitakeshawani** [comment]
> looks good. You could have used uuid for generating id in video.js

Yes but that is the ID of the youtube video that I want to show from youtube.  That is fixed.

File: src/components/sidebar/SideBar.js (hunk 1 above)
Comment (ishitakeshawani) [reply src/components/sidebar/SideBar.js]: done

File: src/components/sidebar/SideBar.js (hunk 2 above)
Comment (ishitakeshawani) [reply src/components/sidebar/SideBar.js]: done

File: src/pages/history/History.js (hunk 3 above)
Comment (ishitakeshawani) [reply src/pages/history/History.js]: done

**Pull request merged**